shelley-ma/formal-spec/value-size.tex

```diff
@@ -1,37 +1,34 @@
-\section{Value Size}
+\section{Output Size}
 \label{sec:value-size}
 
 In Shelley, the protocol parameter $\var{minUTxOValue}$ is used to
 disincentivize attacking nodes by putting many small outputs on the
 UTxO, permanently blocking memory. Because members of the $\Value$
-type can be arbitrary large in principle, the multi-asset ledger must
+type can be arbitrary large, the multi-asset ledger must
 accomodate by scaling the minimum amount of Ada in an output by the
-size of the output, which means that we need a function to determine
-the size of a member of $\Value$.
+size of the output.
```

Comment:
the minimum amount of Ada in an output should be scaled by the size of the *whole UTxO* - the whole "standard" UTxO, containing only coin and no datum), ie input |-> (addr, coin), 
which can vary if we decide to change the lengths of hashes or add some data to the address or whatever. This kind of change should not impact the deposit value, or at least, not implicitly.

This is why i think (now!) it makes sense to have the base `minUTxOValue` be a separate parameter that sets the deposit for the basic type of output. The extra stuff, ie Value type tokens or Datum hash, should have their additional deposit requirements estimated by a separate parameter or parameters, ie

perByteAda :: Coin 
lenDatum :: Integer
lenAssetName :: Integer
OR
perAssetIDAda :: Coin
perDatumAda :: Coin

or something like that

Reply:
I've reworded this to take into account the UTxO keys as well. I don't really see the issue with how the deposit is currently computed, it seems to do exactly what you want.